[Merged by Bors] - feat(analysis/normed_space/affine_isometry): affine maps are open iff their underlying linear maps are

ocfnash opened the pull request:
Formalized as part of the Sphere Eversion project.



---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]
-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

ocfnash committed:
feat(analysis/normed_space/add_torsor): affine maps are continuous or open iff their underlying linear maps are

Formalized as part of the Sphere Eversion project.

ocfnash committed:
Add missing doc string

jcommelin commented on src/analysis/normed_space/add_torsor.lean:
```diff
@@ -301,6 +308,30 @@ end
+variables [semi_normed_space 𝕜 V₂] [pseudo_metric_space P₂] [semi_normed_add_torsor V₂ P₂]
+include V₂
+
+lemma continuous_of_normed_iff (f : P →ᵃ[𝕜] P₂) : continuous f ↔ continuous f.linear :=
```
I would expect a name of the form `continuous_iff_continuous_linear`.

ocfnash replied:
I have just discovered that this lemma already exists under the name `continuous_linear_iff` (in the same `affine_map` namespace) so I have dropped mine https://github.com/leanprover-community/mathlib/blob/fefd8a38be7811574cd2ec2f77d3a393a407f112/src/analysis/normed_space/affine_isometry.lean#L543

jcommelin commented on src/analysis/normed_space/add_torsor.lean:
```diff
@@ -301,6 +308,30 @@ end
+  rw [h, homeomorph.comp_continuous_iff, homeomorph.comp_continuous_iff'],
+end
+
+lemma is_open_map_of_normed_iff (f : P →ᵃ[𝕜] P₂) : is_open_map f ↔ is_open_map f.linear :=
```
Idem

ocfnash replied:
Now called `is_open_map_linear_iff` consistent with extant `continuous_linear_iff`.

jcommelin reviewed (commented):
Comment about the names. Otherwise, LGTM.
Not sure if my suggestion is optimal.

ocfnash committed:
Avoid duplicate lemma but using affine isometry logic.

I had failed to notice `affine_map.continuous_linear_iff` already existed. I
thus need only add the lemma about openness of affine maps.

ocfnash committed:
Flip sides in `affine_map.is_open_map_linear_iff`

ocfnash commented on src/topology/algebra/affine.lean:
```diff
@@ -25,7 +27,8 @@ section ring
-/-- An affine map is continuous iff its underlying linear map is continuous. -/
+/-- An affine map is continuous iff its underlying linear map is continuous. See also
+`affine_map.continuous_linear_iff`. -/
 lemma continuous_iff {f : E →ᵃ[R] F} :
```
@jcommelin raised questions about naming which I have marked as resolved but this old lemma might deserve a rename. It should probably also be restated as `continuous f.linear ↔ continuous f` for consistency with `affine_map.continuous_linear_iff`.

Since I don't actually use this old lemma (I'm just updating its doc string) and I have two other PRs dependent upon this one, I'd vote for carrying out such tidy up as a follow-up PR but of course I'm happy to do whatever others prefer.

PatrickMassot reviewed (commented):
bors merge
